lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/API/Transactions.hs

```diff
@@ -244,101 +294,114 @@ spec = do
+    let (addrD, st') = genChange pwd st
+    addrs <- listAddresses ctx wDest
+    let destination = (addrs !! 1) ^. #id
+    let amt = 100 :: Natural
+    let payload = Json [json|{
                 "payments": [{
-                    "address": #{destination},
-                    "amount": {
-                        "quantity": #{amt},
-                        "unit": "lovelace"
-                    }
-                }],
-                "passphrase": "Secure Passphrase"
-            }|]
-        let (feeMin, _feeMax) = ctx ^. feeEstimator $ TxDescription
-                { nInputs = 1
-                , nOutputs = 1
+                "address": #{destination},
+                "amount": {
+                    "quantity": #{amt},
+                    "unit": "lovelace"
                 }
-        r1 <- request @(ApiTransaction t) ctx (postTxEp wSrc) Default payload
-        verify r1
-            [ expectSuccess
-            , expectResponseCode HTTP.status202
-            ]
-        r2 <- request @ApiWallet ctx (getWalletEp wDest) Default Empty
-        verify r2
-            [ expectSuccess
-            , expectEventually ctx balanceAvailable amt
-            ]
+            }],
+            "passphrase": "Secure Passphrase"
+        }|]
+    let (feeMin, _feeMax) = ctx ^. feeEstimator $ TxDescription
+            { nInputs = 1
+            , nOutputs = 1
+            }
+    r1 <- request @(ApiTransaction t) ctx (postTxEp wSrc) Default payload
+    verify r1
+        [ expectSuccess
+        , expectResponseCode HTTP.status202
+        ]
+    r2 <- request @ApiWallet ctx (getWalletEp wDest) Default Empty
+    verify r2
+        [ expectSuccess
+        , expectEventually ctx balanceAvailable amt
+        ]
+
+    --at this point wDest wallet should have 100 in one of its address
+    --now we try to prepare (tx, [txWintess]), encode it and send 1 to
+    --empty wallet but using external-transactions endpoint
+    txsSrc <- listAllTransactions ctx wSrc
+    let txSrc:_ = filter (\tx ->
+                              any (\(AddressAmount _ amnt) ->
+                                       amnt == Quantity amt)
+                              $ NE.toList $ tx ^. #outputs
+                         ) txsSrc
+    let txHash = getApiT $ txSrc ^. #id
+    let out1@(AddressAmount _ amnt1):out2:_ =
+            NE.toList $ txSrc ^. #outputs
+    let (ix, out) = if amnt1 == Quantity amt then
+            (0::Word32, out1)
+            else
+            (1::Word32, out2)
+    let outAddr = getApiT $ fst $ API.address out
+    let myRnps = [(TxIn txHash ix, TxOut outAddr (Coin 100))]
+    wEmpt <- emptyWallet ctx
+    addrsEmpt <- listAddresses ctx wEmpt
+    let destEmpt = (addrsEmpt !! 1) ^. #id
+    let destAddr = getApiT $ fst destEmpt
+    let left = if amt >= (fromIntegral feeMin) + toSend then
+            fromIntegral $ amt - (fromIntegral feeMin) - toSend :: Word64
+               else
+            0 :: Word64
```

Comment:
This formatting is also odd, because the `if` expression is further to the left than the `if` keyword, and the `else` expression is further to the left than the `else` keyword. 

Would recommend this instead:
```hs
    let left =
            if amt >= (fromIntegral feeMin) + toSend
            then fromIntegral $ amt - (fromIntegral feeMin) - toSend :: Word64
            else 0 :: Word64```
```

Reply:
done